dts/bindings/spi/spi-device.yaml

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright (c) 2018, I-SENSE group of ICCS
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+---
+title: SPI Device Base Structure
+id: spi-device
+version: 0.1
+
+description: >
+    This binding gives the base structures for all spi devices
+
+parent:
+    bus: spi
+
+properties:
+    compatible:
+      type: string
+      category: required
+      description: compatible strings
+    reg:
+      type: array
+      description: Chip select address of device
```

Comment:
Not sure this is right for all SPI modules.
SPBTLE uses GPIO chip select and hence reg has provided here has no meaning.
So, I guess we should keep it (if valid for some modules) but should be optional

Reply:
I used the same approach as linux/Documentation/devicetree/bindings/spi/spi-bus.txt. I plan to add a property cs-gpios in spi.yaml when the PR for the gpio nodes is merged.
If, for example, the SPI has 2 CS lines, and the cs-gpios property looks like this:
```
cs-gpios = <&gpioA 12 0> <&gpioC 11 0>;
```
we have the following mapping:
```
cs0 : &gpioA 12 0
cs1 : &gpioC 11 0
```
The **0** in spi-device@**0** and property reg = <**0**> point to **cs0**.

Reply:
For once, I gave up waiting for my stalling editor to open my linux kernel repo... I should be more patient!
Thanks for the head's up, removing my comment.

Reply:
I always use http://elixir.free-electrons.com/linux/latest/source to browse linux kernel.

Reply:
> I plan to add a property cs-gpios in spi.yaml when the PR for the gpio nodes is merged.

Looking forward to see all bricks working together! This will require some extra modifications in extract_dts_includes.py to be fully functional (specially if we also add stuff like reset-gpios), but should not be to much of a burden